### ISSUE TITLE
Modules imported in support file are not visible through webpack

### ISSUE THREAD

**Issue by gabbersepp** [cypress-io/cypress-webpack-preprocessor#48]
### Current behavior:
Classes that are imported in the support file are loaded but not known to webpack.

### Desired behavior:
Classes should be requireable through webpack.

### How to reproduce:
Please use following minimal working example:
https://github.com/gabbersepp/cypress-missing-module

### Additional Info (images, stack traces, etc)
First I want to explain our use case:
We have a system with several different brands. Each brand can have customized client behaviour. E.g. some brands do have buttons that others don't have.
To be able to test all brands within one cypress testsuite repo, we introduced an environment variable, called "Brand".
And if a brand needs some special behaviour, we create a new class that inherits from the base class and extends the behaviour to reflect the customized behaviour.

Simple example using Typescript (but that does not matter, the same occurs in JS, too. 

```js
export default class SaveLogic {
	...
	public clickSaveButton() {
		cy.get(".default-save-button").click();
	}
	...
}

//.....
export default class SaveLogicBrand1 extends SaveLogic {
	public clickSaveButton() {
		cy.get(".extra-save-button").click();
	}
}
```
Usage:

```js
import SaveLogic from "...";
...
var saveLogic = DependencyResolver.resolve<SaveLogic>(SaveLogic);
saveLogic.clickSaveButton()
...
```

The DependencyResolver knows, based on a anotation of the base class, whether or not it must require the customized class (by naming convention it knows the path).
We use the `__webpack_require__` function as used in the repo.

Using this approach we can use SaveLogic everywhere and have to write our test only once for all but are able to overwrite specific functionality. So we do not reference the brand classes anywhere in the tests.

To overcome the webpack behaviour to include only referenced files, we import all brand files in the cypress support file.

This works. If you execute the example in the repo, the assert `expect(window.asd).to.equal("asd");` is green which means, that Module2.js have been loaded and the field `window.asd` has been set correctly.
But if we try to import that module, webpack does not know it.
See the attached image which shows all modules webpack knows.

Does anyone have a clue how to fix this? Currently our fix is to import the support file in all test suites. 
![webpack_cypress](https://user-images.githubusercontent.com/3234900/55389823-407c6580-5536-11e9-93b6-2ce639003599.JPG)

- Operating System: Windows 10
- Cypress Version: 3.2.0
- Browser Version: Chrome 73.0.x

**Comment by gabbersepp** [comment]
Would be nice, if anyone can tell me whether this is the right repo for this issue. Otherwise I will create the same issue in webpack repo

**Comment by chrisbreiding** [comment]
The support file is bundled and served separately from the spec file. The webpack process and the file created by each is distinct and they don't really know about each other. The only way to share code and functionality between them is through globals. 

While the support file is usually a great way to load code that you need in every spec file, it looks like it won't help you for this use-case. Maybe you can use the [webpack provide plugin](https://webpack.js.org/plugins/provide-plugin/) to auto-import the file you need to require for every spec file, so that you don't have to manually do it?

**Comment by gabbersepp** [comment]
@chrisbreiding I will test the plugin and give you feedback. Thanks for the hint

**Comment by gabbersepp** [comment]
@chrisbreiding This plugin does not help us.
In my example, I had to set something like this:

```
webpack.ProvidePlugin({
    Module2: "xxxxx/xxx/xx/Module2.js"
})

```

But this only works, if I use "Module2" anywhere in the spec file. Otherwise this is ignored because the modules that are configured at this place are only loaded if the given alias is used anywhere.

**Comment by gabbersepp** [comment]
@chrisbreiding  I have added a feature to specify additional entry points. What do you think about this? Tested it successfully with the example at https://github.com/gabbersepp/cypress-missing-module

**Comment by chrisbreiding** [comment]
Released in [4.1.0](https://github.com/cypress-io/cypress-webpack-preprocessor/releases/tag/v4.1.0)
